src/dataset.jl

```diff
@@ -15,6 +15,16 @@ struct KeyedDataset{T<:XArray}
     constraints::OrderedSet{Pattern}
     # Data lookup can be by any type, but typically it'll either be symbol or tuple.
     data::LittleDict{Tuple{Vararg{Symbol}}, T}
+
+    function KeyedDataset(
+        constraints::OrderedSet{Pattern},
+        data::LittleDict{<:Tuple{Vararg{Symbol}}, T},
```

Comment:
```suggestion
        data::LittleDict{<:Tuple{Vararg{Symbol}}, T};
```
Thinking a keyword argument would help identify what the bool is for when called.

Reply:
Since, `check` isn't a very descriptive term, I'm inclined to leave it as a special positional argument. Again, I think if someone really was to pass `false` here then I expect those folks to understand the internals of this package.

Reply:
Ok, given that it's deliberate and `check` is more of an internal feature, I'm ok to leave it.